Evolutionary_loop/Loop_Parts/Part_B/part_b_job2.py

```diff
@@ -0,0 +1,188 @@
+#*******************************************************************************
+# File: part_b_job2.py
+#    Based on Part_B_VPol_job2.sh
+#
+#  Programmer: Jason Yao (yao.966@osu.edu)
+#
+#  Revision history:
+#     07/27/23  Jason Yao, Original version
+#
+#  Notes:
+#       * vertical ruler at column 80
+#  TODO:
+#       1 MOVE SECTION is not great; clean this up
+#       2. p0 & p1 need to be one process, and p0 & p2 need to be another.
+#          (i.e. load xfdtd needs to be absorbed)
+#
+#*******************************************************************************
+'''
+This is Part B2 of the loop, which prepares and runs output.xmacro with the
+relevant parameters (antenna type, population number, grid size, etc.)
+Output.xmacro writes the XFdtd simulation data to output files (.uan)
+The xmacro is mainly made of a xmacro-skeleton text file (see CAT SECTION).
+
+Usage: python3 part_b_job2.py <indiv> <gen> <npop> <working_dir> ... <nsections>
+
+Outout: XmacrosDir/simulation_PEC.xmacro
+
+For more information on the arguments: 
+  >> python3 part_b_job2.py -h
+'''
+
+
+import argparse
+import os
+import subprocess as sp
+from pathlib import Path
+from time import sleep
+
+
+'''
+This function counts the number of files that don't start with "." (hidden) 
+inside a folder "where". Subrirectories are not counted.
+Input: where (Path object)
+Output: counts (int)
+'''
+def count_flags(where=Path):
+    dummy = sp.run(f'cd {where}; ls | wc -l',
+                   shell=True, capture_output=True, text=True)
+    return int(dummy.stdout)
```

Comment:
```suggestion
def count_flags(where: Path):
    return len(list(where.iterdir()))
```

Reply:
Is there an easy way for this to not include hidden files?

Reply:
Something like this:
```suggestion
def count_flags(where: Path):
    return sum(1 for file in where.iterdir() if not file.startswith('.'))
```
Keep in mind this will also count dir, so you might want to do:
```suggestion
def count_flags(where: Path):
    return sum(1 for p in where.iterdir()
               if p.is_file() and not p.startswith('.'))
```

Reply:
I keep getting the following error:
`AttributeError: 'PosixPath' object has no attribute 'startswith'`